app/controllers/detainee_details_controller.rb

```diff
@@ -44,17 +34,41 @@ def update
 
   private
 
-  def redirect_to_move_or_profile
+  def redirect_if_detainee_exists
```

Comment:
Unfortunately, you need to assign the `@_detainee` variable in order to update this very same variable which is set in the in the controller that detainee_details_controller inherits from.